lib/dao.js

```diff
@@ -52,6 +52,45 @@ function getIdValue(m, data) {
   return data && data[idName(m)];
 }
 
+function copyData(from, to) {
+  for (var key in from) {
+    to[key] = from[key];
+  }
+}
+
+function convertDataPropertiesByType(data) {
+  var typedData = {};
+  for (var key in data) {
+    // Convert the properties by type
+    typedData[key] = data[key];
+    if (typeof typedData[key] === 'object'
+            && typedData[key] !== null
+            && typeof typedData[key].toObject === 'function') {
+      typedData[key] = typedData[key].toObject();
+    }
+  }
+  return typedData;
+}
+
+DataAccessObject.prototype.strictChk = function strictChk(strict, data, inst, cb) {
+  var Model = this.constructor;
+  var props = Model.definition.properties;
+  var keys = Object.keys(data);
+  var result = {}, key;
+  for (var i = 0; i < keys.length; i++) {
+    key = keys[i];
+    if (props[key]) {
+      result[key] = data[key];
+    } else if (strict === 'throw') {
+      cb(new Error('Unknown property: ' + key));
+      return;
+    } else if (strict === 'validate') {
+      inst.__unknownProperties.push(key);
+    }
+  }
+  cb(null, removeUndefined(result));
```

Comment:
I find it suspicious that we call `removeUndefined` here, it's not what I would expect from a function handling "strict" model. Also when "strict" mode is off, `removeUndefined` is not called at all, which looks like a bug to me. Could you please check? I would expect that `removeUndefined` is either called in all cases, regardless of "strict" value, or never.

@raymondfeng do you happen to know what's the correct behaviour?